### ISSUE TITLE
recently used doesn't work

### ISSUE THREAD

**Issue by sashei** [mozilla-lockwise/lockwise-ios#416]
steps to reproduce:
- copy a username from an item
- navigate back to entries list
- sort by recently used

expected: item with copied username appears at the top
actual: not necessarily

**Comment by devinreams** [comment]
@sashei To better understand this: you mean the "A-Z" sorter on list view does not have actual recency data behind it to actually sort on?

**Comment by sashei** [comment]
@devinreams yes, I just haven't seen it behave as expected. haven't dug into the root cause yet so I don't know if it's a data problem or just something that's broken in another way.

**Comment by devinreams** [comment]
I've been using the app with "real" data on the TestFlight build and it appears to sort recently for me.

Alpha order for me starts with: 192.168.***, accounts.firefox.com, accounts.google.com...

Recent order then changes to: ally.com, alaskaair.com, southwest.com (I did banking and looked at flights the other day)

So... it seems to be working as expected for me. But perhaps a more rigorous test run / definition is needed.

**Comment by joewalker** [comment]
FWIW, I'm using Lockbox synced with my live data, and I'm fairly sure the "Recent" sort order is not correct.
For example: mozilla.auth0.com, ..., digg.com, ..., google.com
Obviously I've not used digg in the past 100 years. Presumably we should have most recent usage first?

**Comment by joewalker** [comment]
> > I'm fairly sure the "Recent" sort order is not correct.

> Keeping in mind the "recency" is per-device (not going to match your desktop or Firefox mobile used timestamp)...

So where is the sort order correct?

**Comment by devinreams** [comment]
>So where is the sort order correct?

I think it's correct, it's just also confusing. I think what we see in Lockbox is:

1. recent item I copied a password from recently (has recency data)
2. recent item I added a few days ago (has recency data)
3. no activity item (no data)
4. no activity item (no data)
5. ...

And it's not obvious where the "recency" items stop and the "everything else" items begin... does that seem to match what others are seeing and feeling? I agree its weird in early-use but now my "recent data" is starting to fill in and the sorting looks "more useful" to me.

**Comment by joewalker** [comment]
I guess for the people that sign up and connect to Sync, when they first start they have no recency data so the list is confusing. Maybe we should hide the sort-order dropdown for cases where anything other than A-Z is not going to make sense?

**Comment by joeyg** [comment]
@sashei @devinreams I can’t find where Lockbox is calling the DataStore.Touch action. That could be the reason recently used isn’t acting right. What should bump a credential to the top? Copy? Viewing details?
